internal/consensus/replay.go

```diff
@@ -435,7 +438,10 @@ func (h *Handshaker) ReplayBlocks(
 			mockApp := newMockProxyApp(finalizeBlockResponse)
 			h.logger.Info("Replay last block using mock app")
 			state, err = h.replayBlock(state, storeBlockHeight, mockApp)
-			return state.AppHash, err
+			if err != nil {
```

Comment:
FYI, this makes the code cleaner, but the error handling was present before.

Reply:
this was my point to make the code cleaner

Reply:
like we have it here

https://github.com/cometbft/cometbft/blob/9e854cb7f45da5a95df07e9acb517ed8aa24e5af/internal/consensus/replay.go#L499